bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/AbstractZkLedgerManager.java

```diff
@@ -245,15 +245,15 @@ public void process(WatchedEvent event) {
 
     @Override
     public void createLedgerMetadata(final long ledgerId, final LedgerMetadata metadata,
-            final GenericCallback<Void> ledgerCb) {
+            final GenericCallback<LedgerMetadata> ledgerCb) {
         String ledgerPath = getLedgerPath(ledgerId);
         StringCallback scb = new StringCallback() {
             @Override
             public void processResult(int rc, String path, Object ctx, String name) {
                 if (rc == Code.OK.intValue()) {
                     // update version
                     metadata.setVersion(new LongVersion(0));
-                    ledgerCb.operationComplete(BKException.Code.OK, null);
+                    ledgerCb.operationComplete(BKException.Code.OK, metadata);
```

Comment:
So this is the core of the change. I was missing it in the diff.

Reply:
It doesn't really do anything right now, as it's just returning what was passed in. Once immutable metadata is there, this will be something like.

```
LedgerMetadata writtenMetadata = LedgerMetadataBuilder.from(metadata).withVersion(new LongVersion(0)).build();
ledgerCb.operationComplete(BKException.Code.OK, writtenMetadata);
````

Reply:
There's a load of boring changes for this. I'm trying to split out into smaller patches to make the eventually meaty changes easier to review.

Reply:
Sure, I see. I just marked this line as it is the only one which does something really useful in this change.

Reply:
There's also one in the MSLedgerManager, but ya, that's the only slightly interesting part

Reply:
Oh, there are a couple of similar lines.
Please go ahead. These were mostly comments for myself.